src/switch_rtp.c

```diff
@@ -1703,9 +1708,68 @@ static void normalised_ts_commit_dtmf_ts(switch_rtp_t *rtp_session, uint32_t ts)
 	rtp_session->ts_normalised.dtmf_ts = ts;
 }
 
+static void do_cumulative_pp_mos(switch_rtp_t *rtp_session)
+{
+	double R = 0;
+	double mos = 0;
+
+	rtp_session->stats.inbound.cumulative_R_count++;
+	rtp_session->stats.inbound.cumulative_R_total += rtp_session->stats.inbound.pp_R;
+	
+	R = (rtp_session->stats.inbound.cumulative_R_total / rtp_session->stats.inbound.cumulative_R_count);
+	if (R < 0) {
+		R = 0.0;
+	} else if (R > 100) {
+		R = 100;
+	}
+	
+	mos = 1.0 + (0.035) * R + (.000007) * R * (R-60) * (100-R);
+
+	rtp_session->stats.inbound.cumulative_R = R;
+	rtp_session->stats.inbound.cumulative_mos = mos;
+}
+
+static void do_pp_mos(switch_rtp_t *rtp_session)
+{
+	// RTT is calculated when RTCP is enabled
+	double rtt = (rtp_session->rtcp_sock_output && rtp_session->flags[SWITCH_RTP_FLAG_ENABLE_RTCP]) ? rtp_session->stats.rtcp.rtt_avg : 0;
+	// Take the average latency, add jitter, but add an impact to jitter
+	// then add 10 for protocol latencies
+	double R = (rtt/1000 + rtp_session->stats.inbound.variance * rtp_mos_jitter_penalty + 10);
+	unsigned int packet_loss = (rtp_session->stats.rtcp.fraction_lost * 100 / 256);
+	double mos = 0;
+	
+	// Implement a basic curve - deduct 4 for the R value at 160ms of latency
+	// (round trip).  Anything over that gets a much more agressive deduction
+	if (R < 160) {
```

Comment:
Shouldn't this be `<=` condition statement if 160ms is the threshold for tolerated latency

Reply:
Regardless of the formula the deduction would still be 4
for R <= 160
```
R = 100 -160 / 40.0
R = 100 - 4
R = 96
```
For R < 160
```
R = 100 - (160-120) / 10.0
R = 100 - (40) / 10
R = 100 - 4
R = 96
```